### PR TITLE
Use Microsoft.NET.Sdk and iOS/Android .NET Workloads

### PR DESCRIPTION
**Pull request by jonathanpeppers**
* Bump to .NET 5.0.100-rtm.20509.5
* Bump `$(_DotNetPackageVersion)` to 5.0.0-rc.2.20475.5
* Provision iOS/Android workload installers
  * Used the `boots` .NET global tool for this:
    https://github.com/jonathanpeppers/boots
* Microsoft.iOS/Android.Sdk -> Microsoft.NET.Sdk
* Convert Xamarin.Forms app to a single, multi-targeted project.
  * A prototype of the default item groups / wildcards are in
    `Directory.Build.props`
  * Moved logic from `Directory.Build.targets` to
    `HelloForms/Directory.Build.props`.

### CODE REVIEW

File: HelloForms/iOS/Info.plist [hunk 1]
```diff
@@ -21,7 +21,7 @@
         <string>UIInterfaceOrientationLandscapeRight</string>
     </array>
     <key>MinimumOSVersion</key>
-    <string>8.0</string>
+    <string>11.0</string>
```
Comment (rolfbjarne) [inline HelloForms/iOS/Info.plist:24]: Was this needed in the end?

**Review by rolfbjarne — APPROVED** [review]

File: HelloForms/iOS/Info.plist (hunk 1 above)
Comment (jonathanpeppers) [reply HelloForms/iOS/Info.plist:24]: I don't think it's needed, but I can make both iOS projects 7.0 if that is the real minimum.

**Comment by jonathanpeppers** [comment]
The current error is weird, I am not getting it locally:
```
/Users/runner/.dotnet/sdk/5.0.100-rtm.20509.5/NuGet.targets(131,5): Invalid restore input. Duplicate frameworks found: 'net5.0, net5.0'. Input files: /Users/runner/work/1/s/HelloForms/HelloForms.csproj. [/Users/runner/work/1/s/net6-samples.sln]
```
I can `dotnet build net6-samples.sln` just fine on my Mac...

File: global.json [hunk 2]
```diff
@@ -1,6 +0,0 @@
-{
```
Comment (akoeplinger) [inline global.json:1]: you could add this to force people to use the right SDK version:

```json
{
  "sdk": {
    "version": "5.0.100-rtm.20509.5",
    "rollForward": "major",
    "allowPrerelease": true
  }
}
```

**Review by pjcollins — APPROVED** [review]

**Pull request merged**